packages/core/core-flows/src/product/workflows/create-products.ts

```diff
@@ -58,13 +58,17 @@ export const createProductsWorkflow = createWorkflow(
 
       data.createdProducts.forEach((product, i) => {
         const inputProduct = data.input.products[i]
+        let rank = 0
```

Comment:
I would rather move this responsibility to the module as it can be nicely abstracted away